### PR TITLE
test: start using @nextcloud/cypress

### PR DESCRIPTION
**Pull request by max-nextcloud**
### 📝 Summary

Use `@nextcloud/cypress` to enable code sharing with other apps.

For now just using the new fast login method.
This reduces the time it takes to login by 6 sec. on my machine.

* See also: #2201

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/master/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

**Comment by skjnldsv** [comment]
I would migrate the cypress config files to typescript.
Or add cypress to eslint ignore.

https://github.com/import-js/eslint-plugin-import/issues/2495

File: cypress/support/commands.js [hunk 1]
```diff
@@ -21,24 +21,19 @@
...
-	cy.session(user, function() {
-		cy.visit(route)
-		cy.get('input[name=user]').type(user)
-		cy.get('input[name=password]').type(password)
-		cy.get('form[name=login] button[type=submit]').click()
-		cy.url().should('include', route)
-	})
-	// in case the session already existed but we are on a different route...
+Cypress.Commands.add('login', (userId, password, { route, onBeforeLoad } = {}) => {
+	route = route || '/apps/files'
```
Comment (skjnldsv) [inline cypress/support/commands.js:35]:
```suggestion
Cypress.Commands.add('login', (userId, password, { route =  '/apps/files', onBeforeLoad } = {}) => {
```

File: cypress/support/commands.js [hunk 2]
```diff
@@ -21,24 +21,19 @@
...
+	cy.loginWithoutVisit({ userId, password })
 	cy.visit(route, { onBeforeLoad })
```
Comment (skjnldsv) [inline cypress/support/commands.js:37]: I would personally do it the other way around, `loginAndVisit`, and `login` :)

Comment (max-nextcloud) [reply cypress/support/commands.js:37]: I agree... actually i prefer these to be two commands. But I wanted to check if i could just use `@nextcloud/cypress`s login command without touching all the tests.

**Comment by juliushaertl** [comment]
CI seems to take the usual time but not sure why there is no difference

File: cypress.config.js [hunk 3]
```diff
@@ -25,4 +25,5 @@ module.exports = defineConfig({
 		// do not retry in `cypress open`
 		openMode: 0,
 	},
+	"numTestsKeptInMemory": 1,
```
Comment (skjnldsv) [inline cypress.config.js:28]: Oooof, we should all reduce that number, yes! :see_no_evil:

Comment (max-nextcloud) [reply cypress.config.js:28]: I had issues with heap overflows when testing this. Problem also is heap does not get cleaned up when rerunning a test or moving to the next.

This also means loosing the ability to inspect previously run tests - but the benefit still outweights the cost for me here.

I'll still increase this to 5. One test typically takes up 40 MB of heap space. So 5 are 200 MB - so heap is full after 10 test runs.

**Comment by max-nextcloud** [comment]
> CI seems to take the usual time but not sure why there is no difference

Yes. We only log in once per test spec. Afterwards the cookies are reused by the session managment. So even if you have a 3 minute test spec, this will only cut off 6 seconds - which is lost in the noise.

It's more relevant when working on a single test maybe even using `it.only` - because if your test takes only 15 seconds those 6 seconds are a big difference.

**Comment by skjnldsv** [comment]
> It's more relevant when working on a single test maybe even using `it.only` - because if your test takes only 15 seconds those 6 seconds are a big difference.

We have a lot of tests in viewer too, there was a lot of login in and out :)
That should explain why there was such a drastic change there

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**